From 6068f2dc7dfd5b0587c94fa235875a61197409f1 Mon Sep 17 00:00:00 2001
From: Jason Wu <jason.wu@petalogix.com>
Date: Thu, 1 Nov 2012 20:15:56 +1000
Subject: [PATCH] net: ethernet: axi_emac: Fix compilation error with Zynq

Currently Zynq is not aware of in_be32 and out_be32 functions. So if
axi_emac driver is enabled, it will get compilation error.
e.g implicit declaration of function 'in_be32'
	[-Werror=implicit-function-declaration]

Signed-off-by: Jason Wu <huanyu@xilinx.com>
---
 drivers/net/ethernet/xilinx/xilinx_axienet.h |    8 ++++++++
 1 files changed, 8 insertions(+), 0 deletions(-)

diff --git a/drivers/net/ethernet/xilinx/xilinx_axienet.h b/drivers/net/ethernet/xilinx/xilinx_axienet.h
index ee73048..bae70c8 100644
--- a/drivers/net/ethernet/xilinx/xilinx_axienet.h
+++ b/drivers/net/ethernet/xilinx/xilinx_axienet.h
@@ -338,6 +338,14 @@
 
 #define DELAY_OF_ONE_MILLISEC		1000
 
+/* Read/Write access to the registers */
+#ifndef out_be32
+#ifdef CONFIG_ARCH_ZYNQ
+#define in_be32(offset)		__raw_readl(offset)
+#define out_be32(offset, val)	__raw_writel(val, offset)
+#endif
+#endif
+
 /**
  * struct axidma_bd - Axi Dma buffer descriptor layout
  * @next:         MM2S/S2MM Next Descriptor Pointer
-- 
1.7.1

